### ISSUE TITLE
Revive or enable way to report D401 reason

### ISSUE THREAD

**Issue by svenevs** [PyCQA/pydocstyle#360]
I like D401, it makes me create better one-liners.  Sometimes it would be really helpful to know what exactly is triggering the warning.

If you search around the issues here you will find #68 and then #235 which will let you know you should reference the data files added, but this doesn't always help (or maybe it is just confusing to me).

```py
def set_env(**kwargs: str):
    """
    Function decorator to set environment variables.
    """
    pass
```

It can be reversed, e.g. `Set environment variables function decorator.`  So is it mad about the `to`?  You can make a not so great `To set environment variables function decorator.` though.  It's commented out in the blacklist for what I thought is this use case.

The idea would be maybe to:

1. Keep the current generic "try rephrasing" message by default.
2. Enable some way to ask for more info.  E.g., `--verbose` mode or some kind of `setup.cfg` setting or something.  I don't know how this would be achieved though.

Given that I don't think anybody intends to "fully implement" D401 (it's a _very_ hard problem...), there's a bit of fuzz room.  This fuzz room can lead to a lot of time spent trying to reorder things for people like me who aren't great with grammar xD  If I knew what was triggering D401, I could know which part of the sentence needs to be rephrased.

I'm using this indirectly through `flake8-docstrings`, but I think this repo is the right place to discuss.  I'm happy to implement if desired / with a little guidance on how to approach it :slightly_smiling_face:

**Update**: it seems that `function` is the problem child

https://github.com/PyCQA/pydocstyle/blob/2549847f9efad225789f931e83dfe782418ca13e/src/pydocstyle/data/imperatives_blacklist.txt#L49

I would still like to implement a way to report back why the D401 exists.  Maybe `function` should be moved to its friends at the bottom that are marked OK (for situations like this)?

**Comment by lordmauve** [comment]
`function` is in the blacklist because people write

```python
def square(x):
    """Function to square a number."""
```

rather than

```python
def f(x):
    """"Square a number."""
```

**Comment by svenevs** [comment]
Fair enough, I'll just `# noqa` things that are blacklisted for situations like this. Now that I know where to look it's not too challenging to figure out what the problem is :slightly_smiling_face:

I think it's better to leave it blacklisted to prevent the more common case you described. Decorators aren't uncommon, but they're less common by comparison.
